### PR TITLE
CodeUtilities: Add Unit Testing

### PR DESCRIPTION
**Pull request by Windchild292**
Commit 1 adds unit tests, commit 2 expands Javadocs and fixes a rounding bug in MathUtility's long lerp.

### CODE REVIEW

**Review by SJuliez — APPROVED** [review]
All good. Is there a difference between compareNullable and Objects.compare?

**Comment by Windchild292** [comment]
Objects.compare doesn't handle one parameter being null and the other being not null... just both having values or both being null.

File: megamek/src/megamek/codeUtilities/MathUtility.java [hunk 1]
```diff
@@ -21,57 +21,117 @@
...
+     * a f of 0d will return the minimum, a f of 1d will return the maximum. Otherwise, this will
+     * return the rounded long between the two points
+     * @note we assume this is used correctly, and thus do not check if the min and max are handled
+     * incorrectly
      */
-    public static long lerp(final long min, final long max, final long f) {
+    public static long lerp(final long min, final long max, final double f) {
         // The order of operations is important here, to not lose precision
-        return min * (1L - f) + max * f;
+        return Math.round(min * (1L - f) + max * f);
```
Comment (NickAragua) [inline megamek/src/megamek/codeUtilities/MathUtility.java:78]: Every time I see a change to a "deep library" function like this, I get scared. Do we have any idea where it's used? And if so, can we get a short write-up of stuff that can be tested from the user side to see if the change "affects" anything?

Comment (SJuliez) [reply megamek/src/megamek/codeUtilities/MathUtility.java:78]: According to my IDE the lerps are not currently used at all. I guess it's safe to merge.

**Pull request merged**